packages/path_provider/path_provider_macos/example/pubspec.yaml

```diff
@@ -4,18 +4,14 @@ description: Demonstrates how to use the path_provider plugin.
 dependencies:
   flutter:
     sdk: flutter
-  path_provider: ^1.6.14
-
-# path_provider_macos is endorsed, so we need to add it to dependency_overrides
-# to depend on it from path:
-dependency_overrides:
   path_provider_macos:
     # When depending on this package from a real application you should use:
     #   path_provider_macos: ^x.y.z
     # See https://dart.dev/tools/pub/dependencies#version-constraints
     # The example app is bundled with the plugin so we use a path dependency on
     # the parent directory to use the current plugin's version.
     path: ../
+  path_provider_platform_interface: 2.0.0-nullsafety
```

Comment:
Is this how folks should use path_provider, or as you mentioned to remove the circular dependency only?

Reply:
This is to avoid circular dependencies. If someone really wanted to use a platform interface package instead of an app-facing package that's always something they could do, but in general I don't see why someone would want to.

(`example` is arguably a bad name for implementation package testbeds... except that in the case of a plugin that has its own platform-specific API extensions, you would use the implementation package directly, and this would be where the examples of that would be, so it's a bit tricky.)